sw/device/lib/drivers/i2c/i2c.c

```diff
@@ -600,28 +530,94 @@ i2c_result_t i2c_write_byte(const i2c_t *i2c, uint8_t byte,
   // Convert the format code into flags.
   i2c_fmt_flags_t flags = {.suppress_nak_irq = suppress_nak_irq};
   switch (code) {
-    case kDifI2cFmtStart:
+    case kI2cFmtStart:
       flags.start = true;
       break;
-    case kDifI2cFmtTx:
+    case kI2cFmtTx:
       break;
-    case kDifI2cFmtTxStop:
+    case kI2cFmtTxStop:
       flags.stop = true;
       break;
-    case kDifI2cFmtRx:
+    case kI2cFmtRx:
       flags.read = true;
       break;
-    case kDifI2cFmtRxContinue:
+    case kI2cFmtRxContinue:
       flags.read = true;
       flags.read_cont = true;
       break;
-    case kDifI2cFmtRxStop:
+    case kI2cFmtRxStop:
       flags.read = true;
       flags.stop = true;
       break;
     default:
-      return kDifI2cBadArg;
+      return kI2cBadArg;
+  }
+
+  return i2c_write_byte_raw(byte, flags);
+}
+
+
+/****************************************************************************/
+/**                                                                        **/
+/*                            LOCAL FUNCTIONS                               */
+/**                                                                        **/
+/****************************************************************************/
+
+
+uint16_t round_up_divide(uint32_t a, uint32_t b) {
+  if (a == 0) {
+    return 0;
   }
 
-  return i2c_write_byte_raw(i2c, byte, flags);
+  return ((a - 1) / b) + 1;
 }
+
+
+i2c_config_t default_timing_for_speed(i2c_speed_t speed,
```

Comment:
this function relies on multiple ad-hoc values, should we define those in DEFINES etc?